configurator/backend/authorization/service.go

```diff
@@ -63,11 +83,64 @@ func NewService(ctx context.Context, vp *viper.Viper, storage storages.Configura
 		if err != nil {
 			return nil, err
 		}
+
+		ssoAuthProvider = CreateSSOProvider(vp)
 	} else {
 		return nil, errors.New("Unknown 'auth' section type. Supported: firebase, redis")
 	}
 
-	return &Service{authProvider: authProvider, configurationsStorage: storage}, nil
+	return &Service{authProvider: authProvider, ssoAuthProvider: ssoAuthProvider, configurationsStorage: storage}, nil
+}
+
+func CreateSSOProvider(vp *viper.Viper) SSOProvider {
+	var ssoProvider SSOProvider
+	var err error
+
+	ssoConfig := &SSOConfig{}
+	envConfig := os.Getenv("SSO_CONFIG")
+
+	if envConfig == "" {
+		vpSSO := vp.Sub("sso")
+		if vpSSO == nil {
+			logging.Info("No SSO config provided, skipping initialize SSO Auth")
+			return nil
+		}
+
+		ssoConfig = &SSOConfig{
+			Provider:              vpSSO.GetString("provider"),
+			Tenant:                vpSSO.GetString("tenant"),
+			Product:               vpSSO.GetString("product"),
+			Host:                  vpSSO.GetString("host"),
+			AccessTokenTTLSeconds: vpSSO.GetDuration("accessTokenTTLSeconds"),
+		}
+	} else {
+		err = json.Unmarshal([]byte(envConfig), ssoConfig)
+		if err != nil {
+			logging.Errorf("Can't unmarshal SSO_CONFIG from env variables: %v", err)
+			return nil
+		}
+	}
+
+	validate := validator.New()
+	err = validate.Struct(ssoConfig)
+	if err != nil {
+		logging.Errorf("Missed required SSO config params: %v", err)
+		return nil
+	}
+
+	ssoConfig.AccessTokenTTLSeconds *= time.Second
+
+	if ssoConfig.Provider == "boxyhq" {
+		ssoProvider, err = NewBoxyHQProvider(ssoConfig)
+		if err != nil {
+			logging.Error(err)
+			return nil
+		}
+		return ssoProvider
+	} else {
+		logging.Errorf("Provider %s not supported", ssoConfig.Provider)
```

Comment:
It would be nice to have all supported providers in the error message, e.g. : `Provider %s not supported. Jitsu supports only: ....`